Content.Server/Power/Components/BaseCharger.cs

```diff
@@ -226,4 +231,37 @@ private void TransferPower(float frameTime)
             UpdateStatus();
         }
     }
+
+    /// <summary>
+    ///     Fires after entity is inserted into a charger.
+    /// </summary>
+    public sealed class ChargerInsertEvent : EntityEventArgs
+    {
+        /// <summary>
+        ///     InsertedEntity is the actual entity that was put in the charger.
+        /// </summary>
+        public IEntity InsertedEntity { get; }
+
+        public ChargerInsertEvent(IEntity insertedEntity)
+        {
+            InsertedEntity = insertedEntity;
+        }
+    }
+
+    /// <summary>
+    ///     Fires after entity is removed from a charger.
+    /// </summary>
+    public sealed class ChargerEjectEvent : EntityEventArgs
+    {
+        /// <summary>
+        ///     EjectedEntity is the actual entity that was taken out of the charger.
+        /// </summary>
+        public IEntity EjectedEntity { get; }
```

Comment:
Use EntityUid

Reply:
If they use direct events then sending the entity in the first place shouldn't even be required